### ISSUE TITLE
Create a ValueStream<T> from a Stream<T>

### ISSUE THREAD

**Issue by khainke** [ReactiveX/rxdart#410]
TLDR:
It is possible to create a ValueConnectableStream from a Stream<T> like so:
```dart
ValueStream<int> get xSquared => ValueConnectableStream<int>(stream.map((x)=>x*x)); 
```
But this stream will not emit events, until it was connected. Is it possible to create a ValueStream<T> from a Stream<T> that behaves like a BehaviorSubject, meaning it emits events when listened to?

My use case:
I want to use several value streams i've implemented to provide different stream getters for different interface specifications. This might sound complicated, but imagine this, you know in your interface should provide 3 properties (say a,b,c) and in 1 class implementation this might need to be handled seperately at 3 different time points / with 3 different streams, but in another class implementation, these properties might all follow from 1 property / 1 stream (maybe you have a stream with a tuple (a,b,c)). To avoid redundancy, you therefore use 1) combineLatest3 or 2) map to implement the interface. Both will result in a Stream<T>. 
Now for various reasons it would be nice to use ValueStream<T> also in the interface, so that whoever uses the interface also gets supplied with the current value. But after the combining/mapping youre left with a Stream<T>.  With above it is possible to create ConnectableValueStream, but this will not emit events, until connected and the user of the interface will not expect that he has to connect:

So how is it possible to create a ValueStream<T> from a Stream<T>, that starts emitting events when listenend to?

**Comment by hoc081098** [comment]
`shareValue()` or `shareValueSeeded()`

**Comment by khainke** [comment]
thanks
